libsolidity/ast/Types.h

```diff
@@ -398,10 +398,36 @@ class Type
 	mutable std::optional<size_t> m_stackSize;
 };
 
+/**
+ * Base class for Types that can be explicitly converted from another type, which can potentially
+ * lead to a loss in precision.
+ */
+class LossfullyConvertible
```

Comment:
Just a note about naming: I think it's "lossy" rather than "lossful" in this context so I'd rename it to:
```suggestion
class LossilyConvertible
```